store/displayPreferences.ts

```diff
@@ -0,0 +1,224 @@
+import nuxtConfig from '~/nuxt.config';
+
+const stringToBoolean = (value: string) => value === 'True';
+const booleanToString = (value: boolean) => (value ? 'True' : 'False');
+
+export interface DisplayPreferencesState extends DisplayPreferencesDto {
+  CustomPrefs: {
+    [key: string]: string;
+  };
+}
+
+const defaultCustomPrefs: {
+  [key: string]: string;
+} = {
+  darkMode: booleanToString(
+    nuxtConfig.vuetify?.theme?.default === 'dark' || false
+  ),
+  locale: nuxtConfig.i18n?.defaultLocale || 'en'
+};
+
+const defaultState = (): DisplayPreferencesState => ({
+  CustomPrefs: defaultCustomPrefs
+});
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const updateMethods: { [key: string]: any } = {
+  darkMode: (value: string) => {
+    window.$nuxt.$vuetify.theme.dark = stringToBoolean(value);
+  },
+  locale: (value: string) => {
+    window.$nuxt.$i18n.setLocale(value);
+  }
+};
+
+export const state = defaultState();
+
+interface SingleCustomPrefMutationPayload {
+  key: string;
+  value: string;
+}
+
+export const getters: GetterTree<
+  DisplayPreferencesState,
+  DisplayPreferencesState
+> = {
+  /**
+   * Returns the CustomPrefs object
+   *
+   * @param {DisplayPreferencesState} state Current state
+   * @returns {object} CustomPrefs object
+   */
+  getCustomPrefs: (state: DisplayPreferencesState) => state.CustomPrefs,
+
+  /**
+   * Gets the current dark mode setting as a boolean (default is enabled)
+   *
+   * @param {DisplayPreferencesState} state Current state
+   * @returns {boolean} Current dark mode
+   */
+  getDarkMode: (state: DisplayPreferencesState) =>
+    stringToBoolean(state.CustomPrefs.darkMode)
+};
+
+export const mutations: MutationTree<DisplayPreferencesState> = {
+  /**
+   * Sets the internal state with the server answer and assign default custom prefs if not existing
+   *
+   * @param {DisplayPreferencesState} state Current state
+   * @param {any} param1 Payload
+   * @param {DisplayPreferencesDto} param1.displayPreferences Display preferences returned by the server
+   */
+  INIT_STATE(
+    state: DisplayPreferencesState,
+    { displayPreferences }: { displayPreferences: DisplayPreferencesDto }
+  ) {
+    Object.assign(state, displayPreferences);
+    for (const key in defaultCustomPrefs) {
+      if (!(key in state.CustomPrefs)) {
+        state.CustomPrefs[key] = defaultCustomPrefs[key];
+      }
+    }
+  },
+
+  /**
+   * Edits a custom pref key
+   *
+   * @param {DisplayPreferencesState} state Current state
+   * @param {any} param1 Payload
+   * @param {SingleCustomPrefMutationPayload} param1.pref Key and value for new custom pref
+   */
+  EDIT_CUSTOM_PREF(
+    state: DisplayPreferencesState,
+    { pref }: { pref: SingleCustomPrefMutationPayload }
+  ) {
+    state.CustomPrefs[pref.key] = pref.value;
+    if (pref.key in updateMethods) updateMethods[pref.key](pref.value);
+  },
+
+  CALL_ALL_CALLBACKS(state: DisplayPreferencesState) {
```

Comment:
This one should be an `action` that triggers a mutation multiple times

Reply:
Thinking of it, it doesn't even have to be a commit as it doesn't edit the state. Maybe only a dispatch `callAllCallbacks` doing this same loop could do it?

Reply:
I believe so. Try it out

Reply:
Done!